### PR TITLE
LPS-146463 Override isApplicable to disable notifications when the user is not assignable to the workflowtask

### PR DESCRIPTION
**Pull request by seltong**

### CODE REVIEW

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:test:relevant

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]
Hi @seltong, could you add an integration test to cover this method?

Thank you!

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:test:relevant

File: modules/apps/portal-workflow/portal-workflow-task-web/src/test/java/com/liferay/portal/workflow/task/web/internal/notifications/WorkflowTaskUserNotificationHandlerTest.java [hunk 1]
```diff
@@ -143,6 +147,84 @@ public void testInvalidWorkflowTaskIdShouldReturnBlankLink()
 				_serviceContext));
 	}
 
+	@Test
+	public void testIsApplicable() throws Exception {
+		_workflowTaskManager = Mockito.mock(WorkflowTaskManager.class);
+
+		ReflectionTestUtil.setFieldValue(
+			WorkflowTaskManagerUtil.class, "_workflowTaskManager",
+			_workflowTaskManager);
```
Comment (MarinhoFeliphe) [inline modules/apps/portal-workflow/portal-workflow-task-web/src/test/java/com/liferay/portal/workflow/task/web/internal/notifications/WorkflowTaskUserNotificationHandlerTest.java:156]: Hi @seltong,

Get rid of `_workflowTaskManager` variable, please.

Thanks!

File: modules/apps/portal-workflow/portal-workflow-task-web/src/test/java/com/liferay/portal/workflow/task/web/internal/notifications/WorkflowTaskUserNotificationHandlerTest.java [hunk 2]
```diff
@@ -143,6 +147,84 @@ public void testInvalidWorkflowTaskIdShouldReturnBlankLink()
...
+		JSONObject jsonObject = JSONUtil.put(
+			"entryClassName", _VALID_ENTRY_CLASS_NAME
+		).put(
+			"workflowTaskId", _VALID_WORKFLOW_TASK_ID
+		);
+
+		Mockito.when(
+			userNotificationEvent.getPayload()
+		).thenReturn(
+			jsonObject.toJSONString()
```
Comment (MarinhoFeliphe) [inline modules/apps/portal-workflow/portal-workflow-task-web/src/test/java/com/liferay/portal/workflow/task/web/internal/notifications/WorkflowTaskUserNotificationHandlerTest.java:170]: Hi @seltong,

Define this jsonObject inline, please.

Thank you!

File: modules/apps/portal-workflow/portal-workflow-task-web/src/test/java/com/liferay/portal/workflow/task/web/internal/notifications/WorkflowTaskUserNotificationHandlerTest.java [hunk 3]
```diff
@@ -143,6 +147,84 @@ public void testInvalidWorkflowTaskIdShouldReturnBlankLink()
...
+		);
+
+		DefaultWorkflowTask defaultWorkflowTask = new DefaultWorkflowTask();
```
Comment (MarinhoFeliphe) [inline modules/apps/portal-workflow/portal-workflow-task-web/src/test/java/com/liferay/portal/workflow/task/web/internal/notifications/WorkflowTaskUserNotificationHandlerTest.java:173]: Hi @seltong

Could we need this object?

File: modules/apps/portal-workflow/portal-workflow-task-web/src/test/java/com/liferay/portal/workflow/task/web/internal/notifications/WorkflowTaskUserNotificationHandlerTest.java [hunk 4]
```diff
@@ -143,6 +147,84 @@ public void testInvalidWorkflowTaskIdShouldReturnBlankLink()
...
+
+		_serviceContext.setUserId(user1.getUserId());
+
+		boolean applicable = _workflowTaskUserNotificationHandler.isApplicable(
+			mockUserNotificationEvent(
+				_VALID_ENTRY_CLASS_NAME, "Sample Object",
+				_VALID_WORKFLOW_TASK_ID),
+			_serviceContext);
+
+		Assert.assertTrue(applicable);
```
Comment (MarinhoFeliphe) [inline modules/apps/portal-workflow/portal-workflow-task-web/src/test/java/com/liferay/portal/workflow/task/web/internal/notifications/WorkflowTaskUserNotificationHandlerTest.java:207]: Hi @seltong 

Define this assert inline, please.

Thank!

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

File: modules/apps/portal-workflow/portal-workflow-task-web/src/test/java/com/liferay/portal/workflow/task/web/internal/notifications/WorkflowTaskUserNotificationHandlerTest.java (hunk 3 above)
Comment (seltong) [reply modules/apps/portal-workflow/portal-workflow-task-web/src/test/java/com/liferay/portal/workflow/task/web/internal/notifications/WorkflowTaskUserNotificationHandlerTest.java:173]: No, we don't need

**Comment by MarinhoFeliphe** [comment]
ci:close

**Comment by MarinhoFeliphe** [comment]
ci:stop

**Review by MarinhoFeliphe — APPROVED** [review]

**Comment by MarinhoFeliphe** [comment]
ci:forward

**Comment by seltong** [comment]
Resent here #761

**Comment by MarinhoFeliphe** [comment]
Just started reviewing :)

**Pull request closed without merging**